### PR TITLE
[FEATURE] Dense account design

### LINKED ISSUE #3297: [FEATURE] Dense account design
### Please confirm the following:

- [X] I've checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate issues.
- [X] I've requested a single (only one) feature/change in this issue. It complies with the One Request Per GitHub Issue (ORPGI) rule.
- [X] My issue is well-defined and describes how it should be implemented from UI/UX perspective.

### What do you want to be added or improved?

image

It's inefficient to show only 3 accounts on a screen. It will be good if it can show more.

### Why do you need it?

_No response_

### How do you imagine it?

_No response_

### PR DESCRIPTION
**Pull request by shamim-emon**
## Pull request (PR) checklist

Please check if your pull request fulfills the following requirements:

- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [x] I've read and understand the [Developer Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Guidelines.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [x] My PR includes only the necessary changes to fix the issue (i.e., no unnecessary files or lines of code are changed).
- [ ] 🎬 I've attached a **screen recording** of using the new code to the next paragraph (if applicable).

## What's changed?

Describe with a few bullets **what's new:**
- I've updated account design of Accounts tab.

## Risk factors

**What may go wrong if we merge your PR?**
- No known issue.

**In what cases won't your code work?**
- No known case

## Does this PR close any GitHub issues? (do not delete)

- Closes #3297 

## Troubleshooting GitHub Actions (CI) failures ❌
Pull request checks failing? Read our [CI Troubleshooting guide](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md).

### CODE REVIEW

**Review by ILIYANGERMANOV — CHANGES_REQUESTED** [review]
You must not change the default design. Instead create a new one and show if a "Compact accounts" feature flag from `IvyFeatures` is enabled

File: shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt [hunk 1]
```diff
@@ -9,11 +9,17 @@ class IvyFeatures @Inject constructor() : Features {
         key = "sort_categories_alphabetically",
         name = "Sort Categories Alphabetically",
         description = "Sort income and expenses" +
-        " categories alphabetically"
+                " categories alphabetically"
+    )
+    override val showCompactAccounts = BoolFeature(
```
Comment (ILIYANGERMANOV) [inline shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt:14]: rename to `compactAccounts`

File: shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt [hunk 2]
```diff
@@ -9,11 +9,17 @@ class IvyFeatures @Inject constructor() : Features {
...
+        key = "show_compact_accounts",
+        name = "Show Compact Accounts",
```
Comment (ILIYANGERMANOV) [inline shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt:16]: "Compact account UI"

File: shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt [hunk 3]
```diff
@@ -9,11 +9,17 @@ class IvyFeatures @Inject constructor() : Features {
...
+        description = "Loads accounts in Dense Account mode"
```
Comment (ILIYANGERMANOV) [inline shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt:17]: Enables more compact and dense UI for the "Accounts" tab

File: screen/accounts/src/test/java/com/ivy/accounts/AccountsTabPaparazziTest.kt [hunk 4]
```diff
@@ -13,9 +13,16 @@ class AccountsTabPaparazziTest(
     private val theme: PaparazziTheme,
 ) : PaparazziScreenshotTest() {
     @Test
-    fun `snapshot accountTab composable`() {
+    fun `snapshot accountTab nonCompact composable`() {
```
Comment (ILIYANGERMANOV) [inline screen/accounts/src/test/java/com/ivy/accounts/AccountsTabPaparazziTest.kt:16]: You need to record Paparazzi snapshots, see the CI Troubleshooting guide

File: screen/accounts/src/main/java/com/ivy/accounts/AccountsViewModel.kt [hunk 5]
```diff
@@ -51,6 +54,7 @@ class AccountsViewModel @Inject constructor(
     private val totalBalanceWithoutExcluded = mutableStateOf("")
     private val totalBalanceWithoutExcludedText = mutableStateOf("")
     private val reorderVisible = mutableStateOf(false)
+    private val showCompactAccounts = mutableStateOf(false)
```
Comment (ILIYANGERMANOV) [inline screen/accounts/src/main/java/com/ivy/accounts/AccountsViewModel.kt:57]: No need for this state

File: screen/accounts/src/main/java/com/ivy/accounts/AccountsViewModel.kt [hunk 6]
```diff
@@ -81,10 +85,15 @@ class AccountsViewModel @Inject constructor(
             totalBalanceWithExcludedText = getTotalBalanceWithExcludedText(),
             totalBalanceWithoutExcluded = getTotalBalanceWithoutExcluded(),
             totalBalanceWithoutExcludedText = getTotalBalanceWithoutExcludedText(),
-            reorderVisible = getReorderVisible()
+            reorderVisible = getReorderVisible(),
+            showCompactAccounts = getShowCompactAccounts()
         )
     }
 
+    private suspend fun shouldShowCompactAccounts(): Boolean {
```
Comment (ILIYANGERMANOV) [inline screen/accounts/src/main/java/com/ivy/accounts/AccountsViewModel.kt:93]: Delete this

File: screen/accounts/src/main/java/com/ivy/accounts/AccountsViewModel.kt [hunk 7]
```diff
@@ -120,6 +129,11 @@ class AccountsViewModel @Inject constructor(
         return reorderVisible.value
     }
 
+    @Composable
+    private fun getShowCompactAccounts(): Boolean {
+        return showCompactAccounts.value
```
Comment (ILIYANGERMANOV) [inline screen/accounts/src/main/java/com/ivy/accounts/AccountsViewModel.kt:134]: features.compactAccounts.enabledAsState()

File: screen/accounts/src/main/java/com/ivy/accounts/AccountsTab.kt [hunk 8]
```diff
@@ -407,18 +411,115 @@ private fun PreviewAccountsTab(theme: Theme = Theme.LIGHT) {
...
+            id = AccountId(UUID.randomUUID()),
+            name = NotBlankTrimmedString.unsafe("Phyre"),
+            color = ColorInt(Green.toArgb()),
+            asset = AssetCode.unsafe("USD"),
+            icon = null,
+            includeInBalance = true,
+            orderNum = 0.0,
+        )
+
+        val acc2 = com.ivy.data.model.Account(
```
Comment (ILIYANGERMANOV) [inline screen/accounts/src/main/java/com/ivy/accounts/AccountsTab.kt:435]: import Account

File: screen/accounts/src/main/java/com/ivy/accounts/AccountsState.kt [hunk 9]
```diff
@@ -12,5 +12,6 @@ data class AccountsState(
     val totalBalanceWithExcludedText: String,
     val totalBalanceWithoutExcluded: String,
     val totalBalanceWithoutExcludedText: String,
-    val reorderVisible: Boolean
+    val reorderVisible: Boolean,
+    val showCompactAccounts: Boolean
```
Comment (ILIYANGERMANOV) [inline screen/accounts/src/main/java/com/ivy/accounts/AccountsState.kt:16]: rename to compactAccountsModeEnabled, "show" is unnecessary

File: shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt [hunk 10]
```diff
@@ -9,11 +9,17 @@ class IvyFeatures @Inject constructor() : Features {
...
     )
 
     override val allFeatures: List<BoolFeature>
         get() = listOf(
-            sortCategoriesAlphabetically
+            sortCategoriesAlphabetically,
+            showCompactAccounts
```
Comment (ILIYANGERMANOV) [inline shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt:23]: compactAccountsMode

**Review by ILIYANGERMANOV — CHANGES_REQUESTED** [review]
Nice! Starts to look good 💯 Left some comments

File: screen/accounts/src/test/java/com/ivy/accounts/AccountsTabPaparazziTest.kt (hunk 4 above)
Comment (shamim-emon) [reply screen/accounts/src/test/java/com/ivy/accounts/AccountsTabPaparazziTest.kt:16]: Are you referring to Paparazzi's [readme](https://github.com/cashapp/paparazzi) or something else? If something else, it'd be really helpful if you provide me the link of  CI Troubleshooting guide . Thanks in advance.

Comment (shamim-emon) [reply screen/accounts/src/test/java/com/ivy/accounts/AccountsTabPaparazziTest.kt:16]: Capture

Failures are from `FakeTransactionDaoTest`. Are they flakey or caused by changes of this PR?

File: shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt (hunk 1 above)
Comment (shamim-emon) [reply shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt]: Done

File: shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt (hunk 2 above)
Comment (shamim-emon) [reply shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt]: Done

File: shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt (hunk 3 above)
Comment (shamim-emon) [reply shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt]: Done

File: screen/accounts/src/main/java/com/ivy/accounts/AccountsViewModel.kt (hunk 5 above)
Comment (shamim-emon) [reply screen/accounts/src/main/java/com/ivy/accounts/AccountsViewModel.kt]: Done

File: screen/accounts/src/main/java/com/ivy/accounts/AccountsViewModel.kt (hunk 6 above)
Comment (shamim-emon) [reply screen/accounts/src/main/java/com/ivy/accounts/AccountsViewModel.kt]: Done

File: screen/accounts/src/main/java/com/ivy/accounts/AccountsViewModel.kt (hunk 7 above)
Comment (shamim-emon) [reply screen/accounts/src/main/java/com/ivy/accounts/AccountsViewModel.kt]: Done

File: screen/accounts/src/main/java/com/ivy/accounts/AccountsTab.kt (hunk 8 above)
Comment (shamim-emon) [reply screen/accounts/src/main/java/com/ivy/accounts/AccountsTab.kt]: Done

File: screen/accounts/src/main/java/com/ivy/accounts/AccountsState.kt (hunk 9 above)
Comment (shamim-emon) [reply screen/accounts/src/main/java/com/ivy/accounts/AccountsState.kt]: Done

File: shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt (hunk 10 above)
Comment (shamim-emon) [reply shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt]: Done

File: screen/accounts/src/test/java/com/ivy/accounts/AccountsTabPaparazziTest.kt (hunk 4 above)
Comment (ILIYANGERMANOV) [reply screen/accounts/src/test/java/com/ivy/accounts/AccountsTabPaparazziTest.kt:16]: Looks flakiness, do retry fix them? Lately they worked on the CI

Comment (ILIYANGERMANOV) [reply screen/accounts/src/test/java/com/ivy/accounts/AccountsTabPaparazziTest.kt:16]: Feel free to delete those

**Review by ILIYANGERMANOV — APPROVED** [review]
Fix the CI and we can merge

**Comment by shamim-emon** [comment]
> Fix the CI and we can merge

Ok will fix it soon!

File: screen/accounts/src/test/java/com/ivy/accounts/AccountsTabPaparazziTest.kt (hunk 4 above)
Comment (shamim-emon) [reply screen/accounts/src/test/java/com/ivy/accounts/AccountsTabPaparazziTest.kt:16]: Deleted those 2 tests!

**Review by ILIYANGERMANOV — APPROVED** [review]

**Comment by Jack-Works** [comment]
wow, I tried it, it works well, thanks!

**Comment by shamim-emon** [comment]
> wow, I tried it, it works well, thanks!

Welcome

**Pull request merged**